src/platform/nxp/k32w/k32w0/K32W0Config.cpp

```diff
@@ -282,12 +282,31 @@ CHIP_ERROR K32WConfig::WriteConfigValueStr(Key key, const char * str, size_t str
 {
     CHIP_ERROR err;
     PDM_teStatus pdmStatus;
+    uint8_t * pData = NULL;
 
     VerifyOrExit(ValidConfigKey(key), err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND); // Verify key id.
 
     if (str != NULL)
     {
-        pdmStatus = PDM_eSaveRecordData((uint16_t) key, (void *) str, strLen);
+        /* If last byte in string is string termination /0, write as it is,
+         * else add a termination to the string and save to PDM */
+        if (str[strLen - 1] == '\0')
+        {
+            pdmStatus = PDM_eSaveRecordData((uint16_t) key, (void *) str, strLen);
+        }
+        else
+        {
+            pData = (uint8_t *) pvPortMalloc(strLen + 1);
+
+            if (pData != NULL)
+            {
+                memcpy(pData, str, strLen);
+                pData[strLen] = '\0';
+                pdmStatus = PDM_eSaveRecordData((uint16_t) key, (void *) pData, strLen + 1);
+                vPortFree((void *) pData);
+            }
+
+        }
         SuccessOrExit(err = MapPdmStatus(pdmStatus));
     }
     else
```

Comment:
Upper layer should call WriteConfigValueStr only with a NULL terminated string. Otherwise, the upper layer should call WriteConfigValueBin. I suggest removing the check for a NULL terminated string.

Reply:
Removed the check in \0 in latest commit.

Reply:
This is different from what we discussed, here is my understanding:

- if the upper layer calls WriteConfigValue**Str** then it should provide a NULL terminated string. If it does not then it means it has used incorrectly the API and WriteConfigValueStr. That's why I suggested to return error if WriteConfigValue**Str**  is not called with a NULL terminated string.

Reply:
Leaving a snippet of discussion with Boris Zbarsky from Matter development Slack:
[Boris Zbarsky](https://app.slack.com/team/UV5KX3M5Z)  [15 hours ago](https://csamembers.slack.com/archives/G014G30SVV0/p1648143990004079?thread_ts=1648130636.952229&cid=G014G30SVV0)
So in this case the CharSpan has length 2.  So data()[0] and data()[1] are the string bytes, and if you try to look at data()[2] you might get a random value or a segfault

[Boris Zbarsky](https://app.slack.com/team/UV5KX3M5Z)  [15 hours ago](https://csamembers.slack.com/archives/G014G30SVV0/p1648144008271009?thread_ts=1648130636.952229&cid=G014G30SVV0)
If you want to pass to an API that expects a C string with null-termination, then yes, at that point you would need to create one.

So the conclusion is that our adaptation layer needs to handle the data to create the string, as it's not guaranteed that the upper layer sends a string.

Reply:
For reference https://csamembers.slack.com/archives/G014G30SVV0/p1648130636952229